app/views/handlers/confirmation_email.py

```diff
@@ -1,24 +1,44 @@
+import json
+from dataclasses import dataclass
+from datetime import datetime
+from uuid import uuid4
+
+from dateutil.tz import tzutc
 from flask import current_app
-from flask_babel import gettext
+from flask_babel import gettext, lazy_gettext
 
+from app.data_models.session_data import SessionData
+from app.data_models.session_store import SessionStore
 from app.forms.email_form import EmailForm
 from app.globals import get_session_store
 from app.helpers.url_param_serializer import URLParamSerializer
+from app.publisher.exceptions import PublicationFailed
+from app.questionnaire import QuestionnaireSchema
+from app.utilities.schema import load_schema_from_session_data
 from app.views.contexts.email_form_context import build_confirmation_email_form_context
 
 
 class ConfirmationEmailLimitReached(Exception):
     pass
 
 
+class ConfirmationEmailFulfilmentRequestFailed(Exception):
+    pass
+
+
 class ConfirmationEmail:
-    PAGE_TITLE = gettext("Confirmation email")
+    def __init__(self, page_title=None, schema=None):
+        self._session_store: SessionStore = get_session_store()
+        self._session_data: SessionData = self._session_store.session_data
```

Comment:
- I don't think `self._session_data` is necessary
- I think it would be cleaner if `session_store` was passed as an argument to the init